include/item.h

```diff
@@ -75,27 +75,266 @@ typedef struct ItemData {
     u8 padding[2];
 } ItemData;
 
+enum ItemFileType {
+	ITEM_FILE_TYPE_DATA,
+	ITEM_FILE_TYPE_ICON,
+	ITEM_FILE_TYPE_PALETTE,
+	ITEM_FILE_TYPE_GBA,
+};
+
+enum ItemDataParam {
+    ITEM_PARAM_PRICE,
+    ITEM_PARAM_HOLD_EFFECT,
+    ITEM_PARAM_HOLD_EFFECT_PARAM,
+    ITEM_PARAM_PREVENT_TOSS,
+    ITEM_PARAM_IS_SELECTABLE,
+    ITEM_PARAM_FIELD_POCKET,
+    ITEM_PARAM_FIELD_USE_FUNC,
+    ITEM_PARAM_BATTLE_USE_FUNC,
+    ITEM_PARAM_PLUCK_EFFECT,
+    ITEM_PARAM_FLING_EFFECT,
+    ITEM_PARAM_FLING_POWER,
+    ITEM_PARAM_NATURAL_GIFT_POWER,
+    ITEM_PARAM_NATURAL_GIFT_TYPE,
+    ITEM_PARAM_BATTLE_POCKET,
+    ITEM_PARAM_PARTY_USE,
+    ITEM_PARAM_HEAL_SLEEP,
+    ITEM_PARAM_HEAL_POISON,
+    ITEM_PARAM_HEAL_BURN,
+    ITEM_PARAM_HEAL_FREEZE,
+    ITEM_PARAM_HEAL_PARALYSIS,
+    ITEM_PARAM_HEAL_CONFUSION,
+    ITEM_PARAM_HEAL_ATTRACT,
+    ITEM_PARAM_GUARD_SPEC,
+    ITEM_PARAM_REVIVE,
+    ITEM_PARAM_REVIVE_ALL,
+    ITEM_PARAM_LEVEL_UP,
+    ITEM_PARAM_EVOLVE,
+    ITEM_PARAM_ATK_STAGES,
+    ITEM_PARAM_DEF_STAGES,
+    ITEM_PARAM_SPATK_STAGES,
+    ITEM_PARAM_SPDEF_STAGES,
+    ITEM_PARAM_SPEED_STAGES,
+    ITEM_PARAM_ACC_STAGES,
+    ITEM_PARAM_CRIT_STAGES,
+    ITEM_PARAM_PP_UP,
+    ITEM_PARAM_PP_MAX,
+    ITEM_PARAM_PP_RESTORE,
+    ITEM_PARAM_PP_RESTORE_ALL,
+    ITEM_PARAM_HP_RESTORE,
+    ITEM_PARAM_GIVE_HP_EVS,
+    ITEM_PARAM_GIVE_ATK_EVS,
+    ITEM_PARAM_GIVE_DEF_EVS,
+    ITEM_PARAM_GIVE_SPEED_EVS,
+    ITEM_PARAM_GIVE_SPATK_EVS,
+    ITEM_PARAM_GIVE_SPDEF_EVS,
+    ITEM_PARAM_GIVE_FRIENDSHIP_LOW,
+    ITEM_PARAM_GIVE_FRIENDSHIP_MED,
+    ITEM_PARAM_GIVE_FRIENDSHIP_HIGH,
+    ITEM_PARAM_HP_EVS,
+    ITEM_PARAM_ATK_EVS,
+    ITEM_PARAM_DEF_EVS,
+    ITEM_PARAM_SPEED_EVS,
+    ITEM_PARAM_SPATK_EVS,
+    ITEM_PARAM_SPDEF_EVS,
+    ITEM_PARAM_HP_RESTORED,
+    ITEM_PARAM_PP_RESTORED,
+    ITEM_PARAM_FRIENDSHIP_LOW,
+    ITEM_PARAM_FRIENDSHIP_MED,
+    ITEM_PARAM_FRIENDSHIP_HIGH,
+};
+
+/**
+ * @brief Move the item in the corresponding pocket from one slot to another.
+ *
+ * All other items in the pocket will be moved up or down within the pocket,
+ * depending on whether the target slot is above or below the current slot.
+ *
+ * @param pocket    The pocket of items in the bag.
+ * @param srcSlot   The slot of the item to be moved.
+ * @param dstSlot   The slot to which the item will be moved.
+ */
 void Item_MoveInPocket(void *pocket, u16 srcSlot, u16 dstSlot);
-u16 Item_ID(u16 item, u16 type);
+
+/**
+ * @brief Get an ID of the specified type for an item.
+ *
+ * Each type points to a different value for the item's entry in sItemArchiveIDs.
+ *
+ * @param item
+ * @param type      The type of ID to get. 
+ * @return The item's ID of the specified type.
+ */
+u16 Item_ID(u16 item, enum ItemFileType type);
```

Comment:
I feel like this function should be called something more descriptive, but I'm having a hard time coming up with a better name. Maybe `Item_GetIDInItemFile`?

Reply:
How about `Item_FileID`? All the elements accessed here are pointers to a file in a NARC somewhere, with the exception of the GBA IDs.

Reply:
Sounds good to me